gene2phenotype_project/gene2phenotype_app/serializers/panel.py

```diff
@@ -5,6 +5,86 @@
 
 from ..utils import get_date_now
 
+
+class PanelCreateSerializer(serializers.ModelSerializer):
+    """
+        Panel Creation Serializer
+
+        Args:
+            serializers (_type_): 
+                name - A required char field
+                description - An optional char field
+
+
+        Raises:
+            serializers.ValidationError: Raises a validation error when the panel exists
+
+        Returns:
+            _type_: A created panel
+    """    
+    name = serializers.CharField(required=True)
+    description = serializers.CharField(required=False)
+
+    def validate(self, attrs):
+        """
+            Validate the request data 
+
+            Args:
+                attrs (_type_): A dictionary like object containing the request data
+
+            Raises:
+                serializers.ValidationError: Raises a validation error when the panel exists
+
+            Returns:
+                _type_: A validated request object
+        """        
+        name = attrs.get('name')
+        if Panel.objects.filter(name=name).exists():
+            raise serializers.ValidationError({'message': 'Can not create an existing panel'})
+
+        return attrs
+    
+    def create(self, validated_data):
+        """
+            Creation of the panel if panel does not exists
+            Updating the panel if is_visible = 0
+
+            Args:
+                validated_data (_type_): validated request object
+
+            Returns:
+                _type_: Created panel
+        """        
+        name = validated_data.get('name')
+        description = validated_data.get('description', '')
```

Comment:
We have to make the description mandatory too, the website uses both `name` and `description`.
Can you update the User model to not allow null values in description?

Reply:
User or Panel model?

Reply:
Sorry, Panel model.